misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -313,41 +329,45 @@ internal class RealTransacter private constructor(
       }
     }
 
-    private fun parseShard(string: String): Shard {
-      val (keyspace, shard) = string.split('/', limit = 2)
-      return Shard(Keyspace(keyspace), shard)
-    }
-
     override fun <T> target(shard: Shard, function: () -> T): T {
-      return useConnection { connection ->
-        if (connection.isVitess()) {
-          val previousTarget = withoutChecks {
-            // TODO we need to parse out the tablet type (replica or master) from the current target and keep that when we target the new shard
-            // We should only change the shard we're targeting, not the tablet type
-            val previousTarget =
-                connection.createStatement().use { statement ->
-                  statement.executeQuery("SHOW VITESS_TARGET").uniqueString()
-                }
-            connection.createStatement().use { statement ->
-              statement.execute("USE `$shard`")
-            }
+      return if (isVitess()) {
```

Comment:
can this be replaced with ```target(Destination(shard), function)```. "target" body seems to exercise same logic

Reply:
Yes it can but I want to expose `target(shard)` and `replicaRead()` to the end user as I believe those are the two main use cases I want app developers to use. So it's a way to guide them down the right path.